sql: block CDC queries on RLS-enabled tables

spilchen opened the pull request:
Block CDC queries on tables with row-level security (RLS) enabled. Supporting CDC queries on RLS tables requires additional work (see GH issue #142171), so this block is added for 25.2 and will be lifted once full support is implemented.

Epic: CRDB-45203
Release note: none
Closes #139447

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142198#-:-OKS6X5m3RoW-iqweZoF:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aerfrei)

___
*[`-- commits` line 6 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/142198#-OKS6BJZ4L6b1BFUGDwn:-OKS6BJZ4L6b1BFUGDwo:b1qpa7q):*
nit: consider adding `docs-known-limitation` label to #142171 if we don't plan to address it in 25.2.
___
*[`pkg/sql/distsql_plan_changefeed.go` line 386 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/142198#-OKS6M4qAryzL8XSi_lc:-OKS6M4qAryzL8XSi_ld:b-8e4pro) ([raw file](https://github.com/cockroachdb/cockroach/blob/d14eb1487cccec40443874797497993a45316518/pkg/sql/distsql_plan_changefeed.go#L386)):*
> ```Go
> 	// We block tables with row-level security enabled because they can inject
> 	// filters into the select op. This conflicts with the CDC expression's
> 	// expectation that the SELECT contains no filters—that all filters are
> ```

nit: add some spaces in "filters-that".


<!-- Sent from Reviewable.io -->

spilchen committed:
sql: block CDC queries on RLS-enabled tables

Block CDC queries on tables with row-level security (RLS) enabled.
Supporting CDC queries on RLS tables requires additional work (see
GH issue #142171), so this block is added for 25.2 and will be lifted
once full support is implemented.

Epic: CRDB-45203
Release note: none
Closes #139447

spilchen commented:
TFTR!

bors r+
